VAMobile/src/api/auth/getAuthSettings.tsx

```diff
@@ -0,0 +1,44 @@
+import { useQuery } from '@tanstack/react-query'
+
+import { UserAuthSettings } from 'api/types'
+import { checkFirstTimeLogin, deviceSupportedBiometrics, isBiometricsPreferred } from 'utils/auth'
+import { pkceAuthorizeParams } from 'utils/oauth'
+
+import { authKeys } from './queryKeys'
+
+/**
+ * Fetch user Auth Settings
+ */
+const getAuthSettings = async (): Promise<UserAuthSettings> => {
+  const firstTimeLogin = await checkFirstTimeLogin()
+  const supportedBiometric = await deviceSupportedBiometrics()
+  const biometricsPreferred = await isBiometricsPreferred()
+  const { codeVerifier, codeChallenge } = await pkceAuthorizeParams()
+  return {
+    canStoreWithBiometric: !!supportedBiometric,
+    displayBiometricsPreferenceScreen: true,
+    firstTimeLogin: firstTimeLogin,
+    loading: false,
+    loggedIn: false,
+    loggingOut: false,
+    shouldStoreWithBiometric: biometricsPreferred,
+    supportedBiometric: supportedBiometric,
+    syncing: false,
+    codeVerifier: codeVerifier,
+    codeChallenge: codeChallenge,
+  }
+}
```

Comment:
I don't think we should have a single query that manages all of these different values. On nearly all the screens where this is used, `useAuthSettings` is only accessing one thing, e.g. `firstTimeLogin` in the `NotificationManager` or biometrics on the Settings screen. It would also make it easier to see what's going on if they're in separate queries. 

We could have a separate query for all the biometrics related data. And a separate one for `firstTimeLogin`.

`codeVerifier` and `codeChallenge` doesn't need to be stored in state, since they're each only referenced once I believe.

And as mentioned in a previous comment, we should be handling state manually in React Query, so all the state related variables like `loading`, `loggingOut` wouldn't be needed.

All of those changes would basically remove the need for this query

Reply:
code verifier and challenge have been pulled out

Reply:
will pull out biometrics on Tuesday when I am back in. The state management in this is due to the ugly AuthGuard code in the app.tsx file. I agree this shouldn't be that way and have to do some thinking on how to best re write that function. Will also look into first time login then as well